arch/arm/core/aarch32/cortex_m/vector_table.S

```diff
@@ -38,37 +38,45 @@ SECTION_SUBSEC_FUNC(exc_vector_table,_vector_table_section,_vector_table)
 
     .word z_arm_hard_fault
 #if defined(CONFIG_ARMV6_M_ARMV8_M_BASELINE)
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
-    .word z_arm_reserved
+    .word 0
+    .word 0
+    .word 0
+    .word 0
+    .word 0
+    .word 0
+    .word 0
     .word z_arm_svc
-    .word z_arm_reserved
+    .word 0
 #elif defined(CONFIG_ARMV7_M_ARMV8_M_MAINLINE)
     .word z_arm_mpu_fault
     .word z_arm_bus_fault
     .word z_arm_usage_fault
 #if defined(CONFIG_ARM_SECURE_FIRMWARE)
     .word z_arm_secure_fault
 #else
+#if defined(CONFIG_ARMV8_M_SE)
```

Comment:
Maybe we can do `#elif` here?

Reply:
Nope, I don't want to do that because it could be confused that ARMv8_M_SE and ARM_SECURE_FIRMWARE are mutually exclusive, but they are not. In fact, ARM_SECURE_FIRMWARE requires ARMV8_M_SE.

Reply:
Maybe re-organize the #ifdefs here, to have ARMV8_M_SE as external and ARM_SECURE_FIRMWARE as internal ifdef, would be better.

Reply:
done